### DISCUSSION TITLE
I want fix design issue with form wizard

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#8155] [Help] [answered]
Actually its not full width I meant its overlapping Below is my code,
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Wizard::make([
                    Wizard\Step::make('About the role')
                        ->schema([
                            Card::make()
                            ->schema([
                            // Select::make('posted_by')
                            // ->relationship('userid','id'),
                            // TextInput::make('posted_by',auth()->id())->hidden(),

                            TextInput::make('title')
                            ->required()
                            ->autofocus()
                            ->placeholder('Enter a Job Title')
                            ->label('What is the name of the role?'),
                            MarkdownEditor::make('description')
                            ->required()
                            ->label('Please describe the role in 3-5 sentences'),
                            TextInput::make('no_of_openings')
                            // ->required()
                            ->autofocus()
                            ->label('How many professionals are you looking to hire? (Number of Openings)'),
                        ]),
                    ]),
                    Wizard\Step::make('Experience & Skills')
                        ->schema([
                            Select::make('category_or_sector')
                            // ->required()
                            ->relationship('category','name')
                            ->label('skills'),
                            MarkdownEditor::make('additional_requirements')
                            ->label('Additional Requirements'),
                        ]),
                    Wizard\Step::make('Time Commitment & Location')
                        ->schema([
                            Radio::make('type')
                            ->options([
                                'full time' => 'Full Time',
                                'part time' => 'Part Time',
                                'contract' => 'Contract'
                            ])
                            ->label('Job Type'),
                            TextInput::make('duration_number')
                            ->label('Enter a Number'),
                            Select::make('duration_type')
                            ->label('In')
                            ->options([
                                'weeks' => 'Weeks',
                                'months' => 'Months',
                                'yaers' => 'Years'
                            ]),
                            Radio::make('location')
                            ->label('What is location for this role?')
                            ->options([
                                'remote' => 'Remote',
                                'in-person' => 'In Person',
                                'hybrid' => 'Hybrid'
                            ]),
                    ]),
                    Wizard\Step::make('Compensation')
                        ->schema([
                            Fieldset::make('Compensation')
                            ->schema([
                                Radio::make('compensation')
                                ->label('Is this a paid role or a volunteer (unpaid) role?')
                                ->options([
                                    'paid' => 'Paid',
                                    'unpaid' => 'Voulnteer',
                                ])

                            ]),
                    ]),
                    Wizard\Step::make('Additinal Information')
                    ->schema([
                        MarkdownEditor::make('notes')
                            ->label('Notes for Applicants(optional)')
                            ->placeholder('E.g. Any perks/benefits that they might receive during the role?'),

                    ]),

                    Wizard\Step::make('Questions for Applicants')
                    ->schema([
                        MarkdownEditor::make('questions')
                            ->label('Notes for Applicants(optional)')
                            ->placeholder('E.g. why do you fit for this role?'),
                    ]),

                ])->submitAction(new HtmlString('<button type="submit">Submit</button>'))
            ]);
    }

![image](https://github.com/filamentphp/filament/assets/137255082/9a797a2f-1cb3-46aa-9f93-ce85cf63f8c5)
and also If i select paid salary field should be shown else doesnot show,

![image](https://github.com/filamentphp/filament/assets/137255082/342c1515-319a-4712-8891-750359d0a18f)

**Comment by leandrocfe** [comment, marked as answer]
```php
...
    ->submitAction(new HtmlString('<button type="submit">Submit</button>'))
    ->columnSpanFull()
])//or ->columns(1) here
;
```

**Reply by mohandtharsha** [reply]
Thanks Its working fine

**Reply by mohandtharsha** [reply]
Kindly provide solution for the second part of the question like dependent fields when I select paid textbox field should be shown else no

**Reply by leandrocfe** [reply]
```php
Radio::make('compensation')
    ->label('Is this a paid role or a volunteer (unpaid) role?')
    ->options([
        'paid' => 'Paid',
        'unpaid' => 'Voulnteer',
    ])
    ->live(), //reactive() for v2

TextInput::make('input')
    ->visible(fn ($get): bool => $get('compensation') === 'paid')
```

**Reply by mohandtharsha** [reply]
@leandrocfe thank you so much its working

**Reply by mohandtharsha** [reply]
I am facing issue with api i.e, http://127.0.0.1/api/user when I hit with postman its showing like below,
{
    "message": "Unauthenticated."
} If You know the please help me

**Reply by leandrocfe** [reply]
check Laravel docs because it isn't a Filament subject

**Reply by mohandtharsha** [reply]
ok thats fine

**Reply by mohandtharsha** [reply]
Any idea How Do I customize the brandName and user menu items since I am new to filament your help means a lot

**Reply by leandrocfe** [reply]
brandName is the `APP_NAME` variable in your .env file
user menu check the [docs](https://filamentphp.com/docs/3.x/panels/navigation#customizing-the-user-menu)

**Reply by mohandtharsha** [reply]
brandName is the APP_NAME variable in your .env file this one works fine but user menu I am having some issue below is my code,->userMenuItems([
                MenuItem::make()
                    ->label('Change Password')
                    ->url(fn (): string => ChangePassword::getUrl())
                    ->icon('heroicon-o-cog-6-tooth'),
                // ...
            ]); 
below is my changepassword class,<?php

namespace App\Filament\Pages;

use App\Settings\GeneralSettings;
use Filament\Forms;
use Filament\Forms\Form;
use Filament\Pages\SettingsPage;
// use Filament\Widgets\Forms\Components\Button;
use Filament\Forms\Components\TextInput;
use App\Models\Users;

class ChangePassword extends SettingsPage
{
    protected static ?string $model = Users::class;

    protected static ?string $navigationIcon = 'heroicon-o-cog-6-tooth';

    protected static string $settings = GeneralSettings::class;

    public function form(Form $form): Form
    {
        // $data = Users::changePassword();
        // print_r($data);exit;
        return $form
            ->schema([
                TextInput::make('password')
                    ->label('Current Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('current-password'),

                TextInput::make('password')
                    ->label('New Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('new-password'),

                TextInput::make('password')
                    ->label('Confirm New Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('new-password'),

                // Button::make('Change Password')
                //     ->submit(),
            ]);
    }
}
Below is error I am getting
Route [filament.employer.pages.manage-footer] not defined.

**Reply by leandrocfe** [reply]
I believe this error isn't related to the menu item. You need to inspect the code and try to understand what is happening.

**Reply by mohandtharsha** [reply]
Target [Filament\Forms\Contracts\HasForms] is not instantiable while building [Filament\Forms\Form].

I am getting while creating a custom page below is my code,
 
 
 
 <?php

namespace App\Filament\Pages;

use App\Settings\GeneralSettings;
use Filament\Forms;
use Filament\Forms\Form;
use Filament\Pages\SettingsPage;
// use Filament\Widgets\Forms\Components\Button;
use Filament\Forms\Components\TextInput;
use App\Models\Users;

class ChangePassword extends SettingsPage
{
    protected static ?string $model = Users::class;

    protected static ?string $navigationIcon = 'heroicon-o-cog-6-tooth';

    protected static string $settings = GeneralSettings::class;

    public function form(Form $form): Form
    {
        // $data = Users::changePassword();
        // print_r($data);exit;
        return $form
            ->schema([
                TextInput::make('password')
                    ->label('Current Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('current-password'),

                TextInput::make('password')
                    ->label('New Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('new-password'),

                TextInput::make('password')
                    ->label('Confirm New Password')
                    ->required()
                    ->type('password')
                    ->autocomplete('new-password'),

                // Button::make('Change Password')
                //     ->submit(),
            ]);
    }

- }

below is route file,
        <?php
        use Illuminate\Support\Facades\Route;
        use App\Filament\Pages\ChangePassword;
        /*
        |--------------------------------------------------------------------------
        | Web Routes
        |--------------------------------------------------------------------------
        |
        | Here is where you can register web routes for your application. These
        | routes are loaded by the RouteServiceProvider and all of them will
        | be assigned to the "web" middleware group. Make something great!
        |
        */

        Route::get('/', function () {
            return view('welcome');
        });

        Route::get('/employer/change-password', [ChangePassword::class, 'form'])->name('filament.employer.pages.change-password');


  
Below is panelProvider,


        <?php

        namespace App\Providers\Filament;

        use Filament\Http\Middleware\Authenticate;
        use Filament\Http\Middleware\DisableBladeIconComponents;
        use Filament\Http\Middleware\DispatchServingFilamentEvent;
        use Filament\Pages;
        use Filament\Panel;
        use Filament\PanelProvider;
        use Filament\Support\Colors\Color;
        use Filament\Widgets;
        use Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse;
        use Illuminate\Cookie\Middleware\EncryptCookies;
        use Illuminate\Foundation\Http\Middleware\VerifyCsrfToken;
        use Illuminate\Routing\Middleware\SubstituteBindings;
        use Illuminate\Session\Middleware\AuthenticateSession;
        use Illuminate\Session\Middleware\StartSession;
        use Illuminate\View\Middleware\ShareErrorsFromSession;
        use Filament\Navigation\MenuItem;
        use App\Filament\Pages\ChangePassword;


        class EmployerPanelProvider extends PanelProvider
        {
            public function panel(Panel $panel): Panel
            {
                return $panel
                    ->id('employer')
                    ->path('employer')
                    ->login()
                    // ->profile()
                    ->colors([
                        'primary' => Color::Emerald,
                    ])
                    ->discoverResources(in: app_path('Filament/Employer/Resources'), for: 'App\\Filament\\Employer\\Resources')
                    ->discoverPages(in: app_path('Filament/Employer/Pages'), for: 'App\\Filament\\Employer\\Pages')
                    ->pages([
                        Pages\Dashboard::class,
                    ])
                    ->discoverWidgets(in: app_path('Filament/Employer/Widgets'), for: 'App\\Filament\\Employer\\Widgets')
                    ->widgets([
                        Widgets\AccountWidget::class,
                        Widgets\FilamentInfoWidget::class,
                    ])
                    ->middleware([
                        EncryptCookies::class,
                        AddQueuedCookiesToResponse::class,
                        StartSession::class,
                        AuthenticateSession::class,
                        ShareErrorsFromSession::class,
                        VerifyCsrfToken::class,
                        SubstituteBindings::class,
                        DisableBladeIconComponents::class,
                        DispatchServingFilamentEvent::class,
                    ])
                    ->authMiddleware([
                        Authenticate::class,
                    ])
                    ->sidebarCollapsibleOnDesktop()
                    ->userMenuItems([
                        MenuItem::make()
                            ->label('Change Password')
                            ->url(fn (): string => ChangePassword::getUrl())
                            ->icon('heroicon-o-cog-6-tooth'),
                        // ...
                    ]);
            }
        }
        ?>

**Comment by mohandtharsha** [comment]
I have created livewire component by checking filament document But I am get that its throwing below error, Livewire page component layout view not found: [components.layouts.app]
Please suggest an answer
